workitem/expression_compiler.go

```diff
@@ -273,6 +273,15 @@ func (c *expressionCompiler) expressionRefersToJoinedData(e criteria.Expression)
 	return nil, false
 }
 
+// ensureJoinTable make sure the table is joined in the query
+func (c *expressionCompiler) ensureJoinTable(tableName string) {
+	for _, j := range c.joins {
+		if j.TableName == tableName {
+			j.Active = true
```

Comment:
@baijum Do we have to manually activate the join? I thought the expression compiler automatically does it for us.
Can you add some more comments about why do we need this?

Reply:
This _is _ the expression compiler implementation. The `search` package doesn't need to do anything to ensure the table join. I think the comment is good enough.